### ISSUE TITLE
Random Server crash with builder

### ISSUE THREAD

**Issue by oldjollysanta** [ldtteam/minecolonies#3947]
[crash-2019-10-18_21.21.47-server.txt](https://github.com/ldtteam/minecolonies/files/3743696/crash-2019-10-18_21.21.47-server.txt)

### Minecolonies version
Version: minecolonies-0.10.430-ALPHA-universal

### Expected behavior
Starting or finishing a building construction

### Actual behaviour

 
- Game crashes

### Steps to reproduce the problem

Hard to say, doesn't happen every new building or upgrade.
Started happening after I used the scan tool to make a blueprint of a large flat area so I could use as a means to flatten / raise ground in the desert. I queued up 3 plots of land to clear. Once the builder completed the first build task the server crashed, I restart and rejoined and the game crashed again once the builder tried to start the next plot. I restarted and rejoined, tp to the townhall and removed all build tasks asap. The server stopped crashing, but now I get a random crash, only sometimes, when I upgrade or plan a building.

**Comment by oldjollysanta** [comment]
[crash-2019-10-18_21.53.09-server.txt](https://github.com/ldtteam/minecolonies/files/3743832/crash-2019-10-18_21.53.09-server.txt)
[crash-2019-10-18_21.54.07-server.txt](https://github.com/ldtteam/minecolonies/files/3743833/crash-2019-10-18_21.54.07-server.txt)
[crash-2019-10-18_22.09.53-server.txt](https://github.com/ldtteam/minecolonies/files/3743834/crash-2019-10-18_22.09.53-server.txt)
Now my server is crashing every time I try to log in

**Comment by Raycoms** [comment]
Okay, please update to the latest structurize, this will fix it.

**Comment by oldjollysanta** [comment]
I'm already using structurize-0.10.192-ALPHA

**Comment by Raycoms** [comment]
Sorry, my mad, I thought it was in the newest structurize, it's in the newest minecolonies > 432

**Comment by Raycoms** [comment]
Fixed in 5f68b4edcdc0a26de75dc580a90af44c1e1fee56

**Comment by oldjollysanta** [comment]
Ok, nice job!
So just hold out for a new release? Or can I download it now?

**Comment by Raycoms** [comment]
You can find the alpha on curse already. But you have to click "view-all"

**Comment by oldjollysanta** [comment]
Literally just spotted it. Thanks for the help, much appreciated!

**Comment by oldjollysanta** [comment]
I updated to version 435
Was able to log in, but then the server crashes again after a few seconds.
[crash-2019-10-18_22.40.07-server.txt](https://github.com/ldtteam/minecolonies/files/3743938/crash-2019-10-18_22.40.07-server.txt)

**Comment by Raycoms** [comment]
Doesn't seem to be us this time. It's weird though

**Comment by oldjollysanta** [comment]
Ah-ha! I think I fixed it.
I turned it to night (cause I had my suspicion it was a AI issue).
I was able to log in, and stay in without crashes. I started checking over things like requests and inventories. Then I came across the miners hut
![2019-10-18_23 14 39](https://user-images.githubusercontent.com/1407323/67097896-6fbaa180-f1fe-11e9-97a1-20689ca24887.png)
Turns out there a part of his ladder got replaced with a block and he went nuts filling in levels until he hit the top.
I cleared the blocks and fixed the ladder. Fired and rehired the miner (just to be sure he reset) and everything is running fine now.
Weird one..
